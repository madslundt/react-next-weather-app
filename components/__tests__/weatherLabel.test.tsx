import React from "react";
import { shallow } from "enzyme";
import { random } from "faker";

import WeatherLabel from "../weatherLabel";

describe("<WeatherLabel />", () => {
    it("Should render and match snapshot", () => {
        const component = shallow(
            <WeatherLabel label="label" unit="unit" value={1} />
        );

        component.update();

        expect(component).toMatchSnapshot();
    });

    it("Should always show label", async () => {
        const expectedLabel = random.word();

        const component = shallow(<WeatherLabel label={expectedLabel} />);

        const html = component.html();

        expect(html).toContain(expectedLabel);
    });
    it("Should show value if value is greater than 0", async () => {
        const expectedValue = random.number({
            min: 0,
        });

        const component = shallow(
            <WeatherLabel label={""} value={expectedValue} />
        );

        const html = component.html();;

        expect(html).toContain(expectedValue.toString());
    });
    it("Should not show undefined when value is not set", async () => {
        const component = shallow(<WeatherLabel label={""} />);

        const html = component.html();

        expect(html).not.toContain("undefined");
    });
    it("Should show value if value is 0", async () => {
        const expectedValue = 0;

        const component = shallow(
            <WeatherLabel label={""} value={expectedValue} />
        );

        const html = component.html();

        expect(html).toContain(expectedValue);
    });
    it("Should show unit if unit is set", async () => {
        const expectedUnit = random.word();

        const component = shallow(
            <WeatherLabel label={""} value={1} unit={expectedUnit} />
        );

        const html = component.html();

        expect(html).toContain(expectedUnit);
    });
    it("Should not show undefined when unit is not set", async () => {
        const component = shallow(<WeatherLabel label={""} value={1} />);

        const html = component.html();

        expect(html).not.toContain("undefined");
    });
    it("Should not show unit when value is undefined", async () => {
        const expectedUnit = random.word();

        const component = shallow(
            <WeatherLabel label={""} unit={expectedUnit} />
        );

        const html = component.html();

        expect(html).not.toContain(expectedUnit);
    });
    it("Should show unit when unit is set and value is 0", async () => {
        const expectedUnit = random.word();

        const component = shallow(
            <WeatherLabel label={""} value={1} unit={expectedUnit} />
        );

        const html = component.html();

        expect(html).toContain(expectedUnit);
    });
});
