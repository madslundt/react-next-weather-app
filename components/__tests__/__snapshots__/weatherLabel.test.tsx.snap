// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherLabel /> Should render and match snapshot 1`] = `
<span>
  label
  :
  <span
    className="font-bold"
  >
     
    1
     
    unit
  </span>
</span>
`;
